pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -2152,8 +2159,16 @@ static void validateRequest(PinotQuery pinotQuery, int queryResponseLimit) {
       throw new IllegalStateException("SQL query should always have response format and group-by mode set to SQL");
     }
 
+    // throw errors if options is less than 1, rectify if larger that current replicas
+    if (queryOptions.get(Broker.Request.QueryOptionKey.NUM_REPLICA_GROUPS) != null) {
+      Integer numReplicaGroups = QueryOptionsUtils.getNumReplicaGroups(queryOptions);
+      if (numReplicaGroups > numReplicas) {
```

Comment:
If numReplicaGroups is null, you get NPE for unboxing on the next line

Reply:
The behavior has been changed, now it will throw Exception for invalid option